assembler.rb

```diff
@@ -46,6 +45,8 @@ def add_predefined_symbols
   end
 
   def assemble
+    @parser = Parser.new(@input_file)
```

Comment:
`@parser` is now only used within `#assemble`, so if I saw this code for the first time I’d ask “why does it  need to be an instance variable?”. And the answer is: it doesn’t.